### PR TITLE
replace 'detail' key by 'non_field_errors'

### PR DESCRIPTION
**Pull request by RignonNoel**
| Q                                                      | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                      | Fix
| Tickets (_issues_) concerned               | None

---

##### What have you changed ?
Fix `detail` error key by `non_field_errors` key to be more generic and make frontend integration more easy.

##### How did you change it ?
Just change the custom error key in each custom error message.

##### Verification :

**PR standard**
 - [x] Branches naming convention: 
     - `prefix-snake_case_description`
     - ex: `enhancement-create_new_cell`
     - ex: `fix-pep8_standard_on_cell_model`
 - [x] Fill in this automatically generated PR template
 - [x] Do not include issue numbers in the PR title
 - [x] Include screenshots and animated GIFs in your pull request whenever possible

**Code standard**
 - [x] This Pull-Request fully meets the requirements defined in the issue
 - [x] Add or modify the attached tests
 - [x] Follow the Python Styleguide
 - [x] Document new code based on the Documentation Styleguide
 - [x] Use I18N functions when you add some text

### CODE REVIEW

File: source/apiVolontaria/apiVolontaria/tests/tests_view_Users.py [hunk 1]
```diff
@@ -447,7 +447,7 @@ def test_list_users_without_permissions(self):
 
         response = self.client.get(reverse('users'))
 
-        content = {"detail": "You are not authorized to list users."}
+        content = {'non_field_errors': "You are not authorized to list users."}
```
Comment (francisjeanneau) [inline source/apiVolontaria/apiVolontaria/tests/tests_view_Users.py]: Are you sure about this one? Doesn't look like a validation error to me!

File: source/apiVolontaria/apiVolontaria/views.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def get(self, request, *args, **kwargs):
             return self.list(request, *args, **kwargs)
 
         content = {
-            'detail': _("You are not authorized to list users."),
+            'non_field_errors': _("You are not authorized to list users."),
```
Comment (francisjeanneau) [inline source/apiVolontaria/apiVolontaria/views.py]: Same thing here, this looks more like a permission error than a validation error.

File: source/apiVolontaria/apiVolontaria/tests/tests_view_UsersId.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def test_retrieve_user_id_not_exist_without_permission(self):
             )
         )
 
-        content = {"detail": "You are not authorized "
+        content = {'non_field_errors': "You are not authorized "
```
Comment (francisjeanneau) [inline source/apiVolontaria/apiVolontaria/tests/tests_view_UsersId.py]: IDEM

**Comment by RignonNoel** [comment]
Yep i agree with you. I will made a correction ASAP since it's not a problem in the form but in your right.

**Comment by RignonNoel** [comment]
@francisjeanneau This PR is now updated. Some `non_field_errors` need to be more specific and can be directly linked to a specific field but for the moment this PR allow us to have the error in the Frontend.

**Pull request merged**